gsa/ba.go

```diff
@@ -35,33 +35,42 @@ func StrictBorderarray(x string) []int {
 	b := 0
 	i := 1
 	for i < len(x) {
-		if x[b] == x[i] {
+
+		//we have a border
+		if x[i] == x[b] {
 			b++
-			if i == len(x)-1 {
-				ba[i] = b
+			i++
+			//case where last index element is a border
+			if i == len(x) {
+				ba[len(x)-1] = b
+				break
+			}
+			//found border is not strict
+			if x[i] == x[b] {
+				if b-2 >= 0 {
+					b_2 := ba[b-1]
+					//while loop to find greatest strict border (if exists)
+					for b_2 > 0 {
+						if x[i-1] == x[b_2-1] {
+							if x[i] != x[b_2] {
+								ba[i-1] = b_2
+								break
+							}
+						}
+						b_2 = ba[b_2-1]
+					}
```

Comment:
When you have nested loops, I want a proof that the combined loops still run in linear time. If you keep jumping through the border array until you find a strict border, in each iteration of the outer loop, you will have a quadratic time algorithm instead. That's not up to the standards we want for GSA ;)